### PR TITLE
TMA-892: Fix user filters dry run when false

### PR DESCRIPTION
**Pull request by panjan**
Would like to solve it like this in the future: https://jira.intgdc.com/browse/TMA-921

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/81/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/896/ : SUCCESS in 40s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1164/ : SUCCESS in 33s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/93/ : SUCCESS in 12s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/911/ : SUCCESS in 2m 03s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1179/ : FAILURE in 2m 03s

File: lib/gooddata/lcm/actions/synchronize_user_filters.rb [hunk 1]
```diff
@@ -232,10 +233,10 @@ def call(params)
               working_client_ids << client_id
               filters_to_load = GoodData::UserFilterBuilder.get_filters(new_filters, symbolized_config)
               puts "Synchronizing #{filters_to_load.count} filters in project #{project.pid} of client #{client_id}"
-              project.add_data_permissions(filters_to_load, run_params)
+              partial_results = project.add_data_permissions(filters_to_load, run_params)
```
Comment (yenkins) [inline lib/gooddata/lcm/actions/synchronize_user_filters.rb]: Calls 'project.add_data_permissions(filters_to_load, run_params)' 6 times (DuplicateMethodCall)

File: lib/gooddata/models/user_filters/user_filter_builder.rb [hunk 2]
```diff
@@ -387,6 +387,7 @@ def self.execute_mufs(user_filters, options = {})
       end
 
       if dry_run
+        GoodData.logger.warn('Option "dry_run" specified. No user filters will be altered!')
```
Comment (yenkins) [inline lib/gooddata/models/user_filters/user_filter_builder.rb]: Calls 'gooddata.logger' 3 times (DuplicateMethodCall)

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/102/ : SUCCESS in 16s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/919/ : SUCCESS in 2m 09s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1188/ : FAILURE in 2m 01s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/103/ : SUCCESS in 11s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/920/ : SUCCESS in 40s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1189/ : FAILURE in 2m 03s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/104/ : SUCCESS in 12s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/921/ : SUCCESS in 42s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1190/ : SUCCESS in 2m 04s

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-merge-compliance-check/26/ : SUCCESS in 13s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/922/ : SUCCESS in 1m 49s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1191/ : SUCCESS in 39s
- https://ci.intgdc.com/job/gooddata-ruby-integration-tests-zuul/1362/ : SUCCESS in 1h 08m 22s
- https://ci.intgdc.com/job/gooddata-ruby-lcm-integration-tests-zuul/1125/ : SUCCESS in 34m 44s

**Pull request merged**